NitroxServer/Communication/Packets/Processors/PlayerDeathEventProcessor.cs

```diff
@@ -0,0 +1,29 @@
+﻿using NitroxModel.Packets;
+using NitroxModel.Server;
+using NitroxServer.Communication.Packets.Processors.Abstract;
+using NitroxServer.GameLogic;
+
+namespace NitroxServer.Communication.Packets.Processors
+{
+    class PlayerDeathEventProcessor : AuthenticatedPacketProcessor<PlayerDeathEvent>
+    {
+        private readonly PlayerManager playerManager;
+        private readonly ServerConfig serverConfig;
+        public PlayerDeathEventProcessor(PlayerManager playerManager, ServerConfig serverConfig)
+        {
+            this.playerManager = playerManager;
+            this.serverConfig = serverConfig;
+        }
+
+        public override void Process(PlayerDeathEvent packet, Player player)
+        {
+            if(serverConfig.GameMode == "Hardcore")
```

Comment:
Maybe we should modify `ServerConfig` in order to add a way to manipulate `ServerGameMode` enum instead of an hardcoded value